FutureFeatures.md

```diff
@@ -179,6 +179,56 @@ should support:
 * Code unloading capabilities, especially in the context of code garbage
   collection and defragmentation.
 
+## Optimization and JIT library
```

Comment:
Do you mean JIT code generation, or a wasm library which can optimize code, which then gets JIT compiled?

I think @titzer had ideas about where optimizations should be, and where we should assume code resides (I think he wants the wasm VM to be the One Last True Compiler). I'll let him chime in.